### PR TITLE
Detect negated content absent buffer 2224 v12.2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6575

Describe changes:
- detect: unify multi-buffer code

@jasonish what do you think about the template commit ?

Rebase of #10390 

Preliminary work for #10334 

CI seems unhappy with my hack about functions cast... Need to think about it

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 101 | 112 | 110.89% | 

Pipeline 18445

File: src/detect-template-rust-buffer.c [hunk 1]
```diff
@@ -202,3 +160,30 @@ static void DetectTemplateRustBufferRegisterTests(void)
         DetectTemplateRustBufferTest);
 }
 #endif /* UNITTESTS */
+
+void DetectTemplateRustBufferRegister(void)
```
Comment (jasonish) [inline src/detect-template-rust-buffer.c:164]: I think we should keep the convention of unit tests at the end of the file.

Comment (catenacyber) [reply src/detect-template-rust-buffer.c:164]: ok, will fix

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/10422

**Pull request closed without merging**